rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,763 @@
+        properties:
+          type: object
+          additionalProperties:
+            type: string
+        metadataJson:
+          type: string
+        commit:
+          type: boolean
+    PartitionSpec:
+      type: object
+      properties:
+        unpartitioned:
+          type: boolean
+    Schema:
+      type: object
+      properties:
+        aliases:
+          type: object
+          additionalProperties:
+            type: integer
+            format: int32
+    SortOrder:
+      type: object
+      properties:
+        unsorted:
+          type: boolean
+    CreateTableResponse:
+      type: object
+      properties:
+        identifier:
+          $ref: '#/components/schemas/TableIdentifier'
+        tableLocation:
+          type: string
+          description: "Path of of the table just before the standard  \
+            metadata / data folders. This is useful in cases where a different location
+            provider might be used"
+          example: s3://bucket/prod/accounting.db/monthly_sales
+        metadataLocation:
+          type: string
+          description: "Location of the most current primary metadata file for the table"
+          example: s3://bucket/prod/accounting.db/monthly_sales/metadata/00001-0d4ef14f-ef7d-43e7-9af-5f4ad40a1103.metadata.json
+        metadataJson:
+          type: string
+          description: "Stringified JSON representing the tables metadata"
+          example:
+            "TODO"
+        accessToken:
+          type: string
+    SetPropertiesRequest:
+      type: object
+      properties:
+        namespace:
+          type: string
+        properties:
+          type: object
+    ListNamespacesResponse:
+      type: object
+      properties:
+        databases:
+          type: array
+          items:
+            $ref: '#/components/schemas/Namespace'
+    Namespace:
+      type: object
+      description: Reference to one or more levels of a namespace
+      properties:
+        empty:
+          type: boolean
+    GetNamespaceResponse:
+      type: object
+      properties:
+        namespace:
+          $ref: '#/components/schemas/Namespace'
+        properties:
+          type: object
+          additionalProperties:
+            type: string
+    ListTablesResponse:
+      type: object
+      properties:
+        identifiers:
+          type: array
+          items:
+            $ref: '#/components/schemas/TableIdentifier'
+    GetTableResponse:
+      type: object
+      properties:
+        identifier:
+          $ref: '#/components/schemas/TableIdentifier'
+        location:
+          type: string
+        metadataLocation:
+          type: string
+        metadataJson:
+          type: string
+        schema:
+          $ref: '#/components/schemas/Schema'
+        partitionSpec:
+          $ref: '#/components/schemas/PartitionSpec'
+        properties:
+          type: object
+          additionalProperties:
+            type: string
```

Comment:
Does the rest server have to open the metadata.json file to get these?

Reply:
Not necessarily. It could be storing them in a database or even in memory. I would imagine that the rest server should guarantee that it keeps the metadata files up to date when they change, but it doesn't necessarily have to re-read the files if it can be sure that it's information is up to date.